// Matrix/index.js
var _ = require('debit');
var Matrix = _.Extendable.extend('Matrix', {
    constructor: function (supr, db_) {
        var first;
        var db = _.isArray(db_) ? (db_.length ? (_.isArray(db_[0]) ? db_ : [db_]) : db_) : (_.isNumber(db_) ? [
            [db_]
        ] : []);
        this.database = db;
        return this;
    },
    toJSON: function () {
        return this.toArray();
    },
    toArray: function () {
        return this.database;
    },
    eye: function (number_) {
        this.database = eye(number_ || 0);
        return this;
    },
    mapCells: function (fn) {
        this.reduceCells();
    },
    reduceCells: function (fn) {},
    add: function (matrix) {
        if (_isNumber(matrix)) {
            this.mapCells(function (cell) {
                return cell + matrix;
            });
        } else {}
    }
});
module.exports = Matrix;

function eye(number) {
    var i, index = 0,
        database = [];
    while (index < number) {
        i = 0;
        row = [];
        while (i < number) {
            row.push(i === index ? 1 : 0);
            i += 1;
        }
        index += 1;
        database.push(row);
    }
    return database;
}
